src/main.cpp

```diff
@@ -2210,11 +2297,26 @@ bool CheckTxInputs(const CTransaction& tx, CValidationState& state, const CCoins
 
         }
 
+        // nValueIn for a transaction must include TZE inputs.
+        for (unsigned int i = 0; i < tx.vtzein.size(); i++)
+        {
+            const COutPoint &prevout = tx.vtzein[i].prevout;
+            const CCoins *pCoins = inputs.AccessCoins(prevout.hash);
+            assert(pCoins && pCoins->vtzeout.size() > prevout.n);
+
+            // should this check spentness?
```

Comment:
The changes to `HaveInputs` should cover this right?

Reply:
The assertion that `inputs.AccessCoins` returned a valid prevout is inherently a spentness check, since `inputs` is a view of the UTXO sets.